### PR TITLE
Add Canada (HC) label script

### PR DESCRIPTION
**Pull request by KingYiuWoo**
Add script to load Health Canada (HC) drug labels.
python manage.py load_hc_data --type full

### CODE REVIEW

File: dle/data/management/commands/load_hc_data.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+        dl = DrugLabel()  # empty object to populate as we go
+        dl.source = "HC"
+
+        columns = row.find_all("td")
+        # DIN is located at the second column
+        # TODO: Is DIN same as source product number?
+        dl.source_product_number = columns[1].text.strip()
+
+        # Company is located at the third column
+        # TODO: Is Company same as marketer?
```
Comment (ColeDCrawford) [inline dle/data/management/commands/load_hc_data.py:212]: Are these questions for Peter / David?

File: dle/data/management/commands/load_hc_data.py [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+        if dl.link == "":
+            logger.info(f"{dl.product_name} does not have a PDF label: {link_to_drug_details}")
+
+        # If version date is not available, then get the status date instead
+        if dl.version_date == "":
+            for div in divs:
+                p1 = div.find("p", attrs={"class": "col-sm-4"})
+                if p1 and p1.find("strong") and p1.find("strong").text == "Current status date:":
+                    p = div.find("p", attrs={"class": "col-sm-8"})
+                    dl.version_date = p.text
```
Comment (ColeDCrawford) [inline dle/data/management/commands/load_hc_data.py:248]: How clean are the HC version dates? Do we need to use `dateparser`, or are they very standardized?

File: dle/data/management/commands/load_hc_data.py [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+    ]
+
+    # improved initial text parsing step so this clustering problem wasn't so messy
+    def get_fixed_header(self, text):
+        # return center with the lowest edit distance,
+        #   or placeholder (last entry) if no there's good match
+        dists = [levdistance(text.lower(), c.lower()) for c in self.centers]
+        # ix = np.argmin(dists)
+        ix = dists.index(min(dists))
+        if dists[ix] > 0.6 * len(text):
```
Comment (ColeDCrawford) [inline dle/data/management/commands/load_hc_data.py:363]: This says, if the edit distance is too great (`>0.6 * len(text)`), return nothing?

File: dle/data/management/commands/load_hc_data.py [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+                ps = ProductSection(label_product=lp, section_name=key, section_text=text_block)
+                ps.save()
+
+            info["Label Text"] = label_text
+            self.records[product_code] = info
+            logger.info(f"{hc_file} parsed Successfully")
+        except Exception as e:
+            logger.error(self.style.ERROR(repr(e)))
+            logger.error(self.style.ERROR(f"Failed to process {hc_file}, url = {pdf_url}"))
+            self.error_urls[pdf_url] = True
```
Comment (ColeDCrawford) [inline dle/data/management/commands/load_hc_data.py:421]: Let's use the `ParsingError` class to track errors?

File: dle/data/management/commands/load_hc_data.py (hunk 2 above)
Comment (KingYiuWoo) [reply dle/data/management/commands/load_hc_data.py:248]: The version dates are standardized. e.g., https://health-products.canada.ca/dpd-bdpp/info?lang=eng&code=96287

File: dle/data/management/commands/load_hc_data.py (hunk 3 above)
Comment (KingYiuWoo) [reply dle/data/management/commands/load_hc_data.py:363]: yup, this is from Caleb's code. Guess it serves as confidence level or some sort

File: dle/data/management/commands/load_hc_data.py [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+        if import_type == "test":
+            # Test with the first 25 results
+            num_total_results = 25
+        else:
+            # else get the total number of results and parse all of them
+            # Grab the result webpage
+            soup = BeautifulSoup(self.driver.page_source, "html.parser")
+            table = soup.find("table")
+            # TODO investigate why this is sometimes None: AttributeError: 'NoneType' object has no attribute 'get'
+            table_attrs = json.loads(table.get("data-wb-tables"))
```
Comment (ColeDCrawford) [inline dle/data/management/commands/load_hc_data.py:173]: Sometimes this would error on me

**Review by ColeDCrawford — APPROVED** [review]
I think this is good to merge. We could wait until investigating the HC website error (scraping all labels by labels / 25 times - https://cscie599-drug-labels.atlassian.net/browse/DL-192) but that didn't happen the first time I ran it, and doesn't actually kill the script because it only takes ~30s or so per scrape after all the labels and errors are created. So theoretically would finish in an extra 1-2h.

**Pull request merged**